sdk/python/kfp/dsl/_container_op.py

```diff
@@ -296,3 +297,12 @@ def set_retry(self, num_retries: int):
 
   def __repr__(self):
       return str({self.__class__.__name__: self.__dict__})
+
+  def _set_metadata(self, metadata):
+    '''_set_metadata passes the containerop the metadata information
+    Args:
+      metadata (ComponentMeta): component metadata
+    '''
+    if not isinstance(metadata, ComponentMeta):
+      raise ValueError('_set_medata is expecting ComponentMeta.')
+    self._metadata = metadata
```

Comment:
Thanks for making this field private. I think that we need to make most of the `ContainerOp` fields private as well.

Reply:
Yes, this should not be modified by the users.